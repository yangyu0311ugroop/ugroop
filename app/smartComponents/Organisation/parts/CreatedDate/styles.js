const styles = () => ({
  lineIndicator: {
    width: 40,
    height: 2,
    backgroundColor: '#8abd84',
  },
  name: {
    lineHeight: 1.6,
    marginTop: 8,
  },
  createddate: {
    fontSize: 14,
    padding: '8px 0 8px 0',
  },
});

export default styles;
