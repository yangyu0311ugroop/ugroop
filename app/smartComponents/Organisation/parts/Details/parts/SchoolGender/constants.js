const TITLE = 'schoolGender';

export const ORG_FIELD_VARIANTS = {
  TITLE,
};
export const ORG_FORM_NAME = 'type';

export const SCHOOL_GENDERS = [
  { code: 'boys', name: 'Boys School' },
  { code: 'girls', name: 'Girls School' },
  { code: 'coEducational', name: 'Co-Educational' },
];
