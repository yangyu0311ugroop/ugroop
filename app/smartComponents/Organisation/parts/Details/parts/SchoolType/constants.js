const TITLE = 'schooltype';

export const ORG_FIELD_VARIANTS = {
  TITLE,
};
export const ORG_FORM_NAME = 'schooltype';

export const SCHOOL_TYPES = [
  'Primary',
  'Secondary',
  'K-10/12',
  'Special',
  'College',
  'University',
];
