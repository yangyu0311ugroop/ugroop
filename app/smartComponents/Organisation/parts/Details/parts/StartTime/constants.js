const TITLE = 'starttime';

export const ORG_FIELD_VARIANTS = {
  TITLE,
};

export const ORG_FORM_NAME = 'starttime';
