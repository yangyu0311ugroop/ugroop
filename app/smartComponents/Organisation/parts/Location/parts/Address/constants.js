const TITLE = 'address';

export const ORG_FIELD_VARIANTS = {
  TITLE,
};

export const ORG_FORM_NAME = 'address';
