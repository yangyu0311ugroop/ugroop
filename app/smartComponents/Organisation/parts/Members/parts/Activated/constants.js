const TITLE = 'activated';

export const ORG_FIELD_VARIANTS = {
  TITLE,
};

export const ORG_FORM_NAME = 'activated';
