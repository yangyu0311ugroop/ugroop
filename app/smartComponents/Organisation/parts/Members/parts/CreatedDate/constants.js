const TITLE = 'createdAt';

export const ORG_FIELD_VARIANTS = {
  TITLE,
};

export const ORG_FORM_NAME = 'createdAt';
