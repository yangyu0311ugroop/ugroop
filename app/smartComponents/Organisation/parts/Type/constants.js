const TITLE = 'type';

export const ORG_FIELD_VARIANTS = {
  TITLE,
};

export const ORG_FORM_NAME = 'type';
