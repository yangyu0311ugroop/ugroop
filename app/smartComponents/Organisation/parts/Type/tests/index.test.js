import { DO_NOTHING } from 'appConstants';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import React from 'react';
import { ORG_FIELD_VARIANTS } from 'smartComponents/Organisation/constants';
import mockStylesheet from 'utils/mockStylesheet';
import theme from 'theme/coolTheme';
import Form from 'ugcomponents/Form';
import styles from '../styles';
import { Type } from '../index';

const mockStyles = mockStylesheet('OrgType', styles, theme);

describe('<OrgType />', () => {
  let rendered;
  let instance;

  const resaga = {
    setValue: jest.fn(),
    dispatchTo: jest.fn(),
  };

  const intl = {
    formatMessage: jest.fn(),
  };

  const props = {
    classes: mockStyles,
    resaga,
    intl,
    orgTypes: [{ 0: { code: 1, name: 'test' } }],
  };

  beforeEach(() => {
    rendered = shallow(<Type {...props} />);
    instance = rendered.instance();
    jest.clearAllMocks();
  });

  it('should exists', () => {
    expect(Type).toBeDefined();
  });

  it('should render without exploding', () => {
    expect(rendered.length).toBe(1);
  });

  describe('componentDidMount()', () => {
    it('should call getOrgTypes', () => {
      instance.getOrgTypes = jest.fn(() => 'getOrgTypes');

      rendered.setProps({ orgTypes: [] });

      expect(instance.componentDidMount()).toBe('getOrgTypes');
    });

    it('should DO_NOTHING', () => {
      rendered.setProps({ orgTypes: [1, 2, 3] });

      expect(instance.componentDidMount()).toBe(DO_NOTHING);
    });
  });

  describe('getOrgTypes()', () => {
    it('should call resaga.dispatchTo', () => {
      instance.getOrgTypes();

      expect(resaga.dispatchTo).toBeCalled();
      expect(resaga.dispatchTo.mock.calls).toMatchSnapshot();
    });
  });

  describe('renderSelect()', () => {
    it('should renderSelect', () => {
      rendered.setProps({ orgTypes: [1, 2, 3] });

      const snapshot = shallow(<div>{instance.renderSelect()}</div>);

      expect(toJSON(snapshot)).toMatchSnapshot();
    });
  });

  describe('render()', () => {
    it('should render correctly', () => {
      const snapshot = shallow(<div>{instance.render()}</div>);

      expect(toJSON(snapshot)).toMatchSnapshot();
    });
    it('should render text only variant', () => {
      rendered.setProps({
        variant: ORG_FIELD_VARIANTS.TEXT_ONLY,
      });
      const snapshot = shallow(<div>{instance.render()}</div>);

      expect(toJSON(snapshot)).toMatchSnapshot();
    });
    it('should render text field if variant is text field', () => {
      rendered.setProps({
        variant: ORG_FIELD_VARIANTS.TEXT_FIELD,
      });
      const snapshot = shallow(<Form>{instance.render()}</Form>);

      expect(toJSON(snapshot)).toMatchSnapshot();
    });
    it('should render text field if variant is text field', () => {
      rendered.setProps({
        variant: ORG_FIELD_VARIANTS.TEXT_WITH_LABEL,
      });
      const snapshot = shallow(<Form>{instance.render()}</Form>);

      expect(toJSON(snapshot)).toMatchSnapshot();
    });
  });
});
