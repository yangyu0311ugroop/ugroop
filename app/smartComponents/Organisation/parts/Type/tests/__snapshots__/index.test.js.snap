// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrgType /> getOrgTypes() should call resaga.dispatchTo 1`] = `
Array [
  Array [
    "Organisations",
    "getOrgTypes",
    Object {},
  ],
]
`;

exports[`<OrgType /> render() should render correctly 1`] = `
<div>
  <P
    intl={
      Object {
        "formatMessage": [MockFunction],
      }
    }
  >
    Tour Operator
  </P>
</div>
`;

exports[`<OrgType /> render() should render text field if variant is text field 1`] = `
<Fragment>
  <ContextProvider
    value={
      Object {
        "isFormDisabled": [Function],
      }
    }
  >
    <Formsy
      disabled={false}
      getErrorMessage={[Function]}
      getErrorMessages={[Function]}
      getValue={[Function]}
      hasValue={[Function]}
      isFormDisabled={[Function]}
      isFormSubmitted={[Function]}
      isPristine={[Function]}
      isRequired={[Function]}
      isValid={[Function]}
      isValidValue={[Function]}
      mapping={null}
      onChange={[Function]}
      onError={[Function]}
      onInvalid={[Function]}
      onInvalidSubmit={[Function]}
      onReset={[Function]}
      onSubmit={[Function]}
      onValid={[Function]}
      onValidSubmit={[Function]}
      preventExternalInvalidation={false}
      resetValue={[Function]}
      setValidations={[Function]}
      setValue={[Function]}
      showError={[Function]}
      showRequired={[Function]}
      validationErrors={null}
    >
      <Wrapped
        canSubmitForm={true}
        cancelButtonContent="Discard"
        cancelForm={[Function]}
        isFormChanged={false}
        isFormDisabled={false}
        loading={false}
        renderCancelButton={[Function]}
        renderSubmitButton={[Function]}
        resetForm={[Function]}
        submitButtonContent="Submit"
        submitForm={[Function]}
      >
        <Formsy(WithStyles(SelectField))
          innerRef={[Function]}
          intl={
            Object {
              "formatMessage": [MockFunction] {
                "calls": Array [
                  Array [
                    Object {
                      "defaultMessage": "Type",
                      "id": "app.smartComponents.Organisation.parts.Type.label",
                    },
                  ],
                  Array [
                    Object {
                      "defaultMessage": "Type",
                      "id": "app.smartComponents.Organisation.parts.Type.label",
                    },
                  ],
                ],
                "results": Array [
                  Object {
                    "type": "return",
                    "value": undefined,
                  },
                  Object {
                    "type": "return",
                    "value": undefined,
                  },
                ],
              },
            }
          }
          name="type"
          options={
            Array [
              Object {
                "children": undefined,
                "value": undefined,
              },
            ]
          }
          required={false}
          validationError=""
          validationErrors={Object {}}
          validations={null}
          value=""
        />
      </Wrapped>
    </Formsy>
  </ContextProvider>
  <DiscardConfirmationDialog
    onCancel={[Function]}
    onConfirm={[Function]}
    open={false}
  />
</Fragment>
`;

exports[`<OrgType /> render() should render text field if variant is text field 2`] = `
<Fragment>
  <ContextProvider
    value={
      Object {
        "isFormDisabled": [Function],
      }
    }
  >
    <Formsy
      disabled={false}
      getErrorMessage={[Function]}
      getErrorMessages={[Function]}
      getValue={[Function]}
      hasValue={[Function]}
      isFormDisabled={[Function]}
      isFormSubmitted={[Function]}
      isPristine={[Function]}
      isRequired={[Function]}
      isValid={[Function]}
      isValidValue={[Function]}
      mapping={null}
      onChange={[Function]}
      onError={[Function]}
      onInvalid={[Function]}
      onInvalidSubmit={[Function]}
      onReset={[Function]}
      onSubmit={[Function]}
      onValid={[Function]}
      onValidSubmit={[Function]}
      preventExternalInvalidation={false}
      resetValue={[Function]}
      setValidations={[Function]}
      setValue={[Function]}
      showError={[Function]}
      showRequired={[Function]}
      validationErrors={null}
    >
      <Wrapped
        canSubmitForm={true}
        cancelButtonContent="Discard"
        cancelForm={[Function]}
        isFormChanged={false}
        isFormDisabled={false}
        loading={false}
        renderCancelButton={[Function]}
        renderSubmitButton={[Function]}
        resetForm={[Function]}
        submitButtonContent="Submit"
        submitForm={[Function]}
      >
        <WithStyles(ForwardRef(InputLabel))
          shrink={true}
        >
          Type
        </WithStyles(ForwardRef(InputLabel))>
        <P
          noMargin={true}
        >
          Tour Operator
        </P>
      </Wrapped>
    </Formsy>
  </ContextProvider>
  <DiscardConfirmationDialog
    onCancel={[Function]}
    onConfirm={[Function]}
    open={false}
  />
</Fragment>
`;

exports[`<OrgType /> render() should render text only variant 1`] = `
<div>
  <P
    intl={
      Object {
        "formatMessage": [MockFunction],
      }
    }
  >
    Tour Operator
  </P>
</div>
`;

exports[`<OrgType /> renderSelect() should renderSelect 1`] = `
<div>
  <WithStyles(Formsy(InlineRadioGroup))
    autoClose={true}
    color="primary"
    highlightSelected={true}
    label="Organisation type"
    name="type"
    options={
      Object {
        "undefined": undefined,
      }
    }
    required={true}
    tooltip="Click button to change type"
    valueLabel="Organisation type"
  />
</div>
`;
